scenes/enumerated/problem_10349/problem_10349.py

```diff
@@ -0,0 +1,393 @@
+        )
+
+        # Մկրատով կտրենք 10-երը
+        scissors = [
+            *[DScissors(x3_one_part_and_10[i][1].left_edge.get_center())
+              for i in range(len(x3_one_part_and_10))],
+            *[DScissors(x3_one_part_and_10[len(x3_one_part_and_10) - 1 - i][1].right_edge.get_center())
+              for i in range(len(x3_one_part_and_10) - 1)]
+        ]
+        self.play(*[CutIn(scissors[i]) for i in range(len(scissors) - 2)])
+        self.play(*[CutOut(scissors[i]) for i in range(len(scissors) - 2)])
+        self.play(*[CutIn(scissors[i + 3]) for i in range(len(scissors) - 3)])
+        self.play(*[CutOut(scissors[i + 3]) for i in range(len(scissors) - 3)])
+
+        for i in range(len(x3_one_part_and_10)):
+            x3_one_part_and_10[i][1].set_label(Tex(segment_10_string))
+
+        segment_10_labels = [segment_10_label.copy() for _ in range(len(x3_one_part_and_10))]
+
+        self.play(AnimationGroup(
+            *[x3_one_part_and_10[i][1].animate.shift(UP * .5) for i in range(len(x3_one_part_and_10))],
+            *[segment_10_labels[i].animate.next_to(x3_one_part_and_10[i][1], UP, buff=0.7)
+              for i in range(len(x3_one_part_and_10))],
+            lag_ratio=0.2), run_time=1)
+        self.wait()
+
+        # Ձևավոր փակագծեր ամբողջ ճանապարհին
+        entire_road = VGroup(
+            *[x5_one_part[i] for i in range(len(x5_one_part))],
+            *[x3_one_part_and_10[i] for i in range(len(x3_one_part_and_10))],
+        )
+
+        entire_road_brace = Brace(entire_road, DOWN)
+        entire_road_brace_label = Tex(road_length_string).next_to(entire_road_brace, DOWN, buff=0.5)
+        self.play(
+            GrowFromCenter(entire_road_brace),
+            Write(entire_road_brace_label)
+        )
+        self.wait()
+
+        # Ամբողջ ճանապարհից 10-երը հանելու հավասարումը
+        minus_parts = road_len - 3 * int(segment_10_string)
+        x8_part_value = MathTex(f"{road_len}-3\cdot{segment_10_string}={minus_parts}").to_edge(
+            RIGHT, buff=3.5).to_edge(UP, buff=1)
+        self.play(Write(x8_part_value))
+        self.wait()
+        self.play(
+            FadeOut(
+                *[x3_one_part_and_10[i][1] for i in range(len(x3_one_part_and_10))],
+                *[segment_10_labels[i] for i in reversed(range(len(x3_one_part_and_10)))],
+            )
+        )
+
+        # Ամբողջ ճանապարհից 10-երը հանելու ձևավոր փակագծերը
+        x8_one_part = VGroup(*[one_part.copy() for _ in range(8)]).arrange(RIGHT, buff=0).align_to(x5_one_part[0], LEFT+UP)
+        x8_one_part_brace = Brace(x8_one_part, DOWN)
+        x8_one_part_brace_label = MathTex(cut_road_length_string).next_to(x8_one_part_brace, DOWN, buff=0.5)
+        self.play(
+            ReplacementTransform(
+                VGroup(
+                    *[x5_one_part[i] for i in range(len(x5_one_part))],
+                    *[x3_one_part_and_10[i][0] for i in reversed(range(len(x3_one_part_and_10)))]
```

Comment:
Same

Reply:
The second part is nested and can be tricky. Any suggestions?